File: May-Leetcoding-Challenge/16. Odd Even Linked List/OddEvenLinkedList.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode oddEvenList(ListNode head) 
    {
        if (head == null || head.next == null) 
		    return head;
	
        ListNode node1prev = new ListNode(0);
        ListNode node2prev = new ListNode(0);
        ListNode odd = node1prev;
        ListNode even = node2prev;
        
        int count = 1;
        ListNode temp = head;
        
        while(temp != null)
        {
            if(count % 2 == 1)
            {
                odd.next= temp;
                odd = temp;
            }
            else
            {
                even.next= temp;
                even = temp;
            }
            count += 1;
            temp = temp.next;
        }
        even.next = null;
        odd.next = node2prev.next;
        return node1prev.next;    
    }
}

File: May-Leetcoding-Challenge/16. Odd Even Linked List/OddEvenLinkedList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def oddEvenList(self, head: ListNode) -> ListNode:
        if not head or head.next == None:
            return head 
        
        node1prev = odd = ListNode(0)
        node2prev = even = ListNode(0)
        count = 1
        temp = head
        while(temp):
            if count % 2 == 1:
                odd.next, odd = temp, temp
            else:
                even.next, even = temp, temp
            count = count + 1
            temp = temp.next
            
        even.next, odd.next = None, node2prev.next
        return node1prev.next